meta.jq

```diff
@@ -151,63 +133,50 @@ def build_command:
 	normalized_builder as $builder
 	| if $builder == "buildkit" then
 		git_build_url as $buildUrl
-		| (
-			(should_use_docker_buildx_driver | not)
-			or docker_uses_containerd_storage
-		) as $supportsAnnotationsAndAttestsations
 		| [
 			(
 				[
 					@sh "SOURCE_DATE_EPOCH=\(.source.entry.SOURCE_DATE_EPOCH)",
 					# TODO EXPERIMENTAL_BUILDKIT_SOURCE_POLICY=<(jq ...)
 					"docker buildx build --progress=plain",
-					if $supportsAnnotationsAndAttestsations then
-						"--provenance=mode=max",
+					"--provenance=mode=max",
+					if build_should_sbom then
 						# see "bashbrew remote arches docker/scout-sbom-indexer:1" (we need the SBOM scanner to be runnable on the host architecture)
 						# bashbrew remote arches --json docker/scout-sbom-indexer:1 | jq '.arches | keys_unsorted' -c
 						if .build.arch as $arch | ["amd64","arm32v5","arm32v7","arm64v8","i386","ppc64le","riscv64","s390x"] | index($arch) then
```

Comment:
Maybe we should move this to the function `build_should_sbom` it woukld make it easier in the future to add sbom to other arch

Reply:
I thought about that, but it felt wrong given that the function is entirely based on whether the image/build itself should have an SBOM and _this_ check is whether or not the architecture we're building is even supported by our SBOM-generating image (which is a slightly different concern).  This check is also fundamentally wrong to some extent because it needs to be based on the _host_/build system architecture, _not_ the architecture of the image (AFAIK, BuildKit correctly does _not_ apply `--platform` to the SBOM generator image).

Ultimately, I think it'd be better to refactor the way we set `BASHBREW_BUILDKIT_SBOM_GENERATOR` in the script that invokes this such that it can more appropriately make the decision as to whether we should/do support the SBOM generator image it's pointing at on the current architecture, and then this conditional instead becomes `if env.BASHBREW_BUILDKIT_SBOM_GENERATOR then ...`.

However, with the refactoring in this PR, we can _also_ more easily choose to add SBOMs only for specific architectures of a given repository (or even specific variants if we needed/wanted to!), but I think that's an entirely separate concern to the intent of this `if` statement. :+1:

Reply:
Ok, sounds good